giraffe/src/types/index.ts

```diff
@@ -295,6 +299,109 @@ export interface ScatterLayerConfig {
   symbol?: string[]
 }
 
+export interface TableGraphLayerConfig {
+  type: 'table' // do not refactor or restrict to LayerTypes.Table
```

Comment:
This causes code in the storybook demos, which create configs, to look verbose because you can no longer use just an object. By itself, that is not a problem because we are responsible for the code in the storybook demos. However, this also extends to any example snippet. If someone wanted to try the snippets in the READMEs in their own React App, they would get a TypeScript error that the object they are creating for their config is not a TableGraphLayerConfig (or whatever config they may be working on).

In essence, we want example snippets to be painless for 3rd party users who want to try Giraffe in their React Apps. A simple comment in our code warning that it shouldn't be refactored seems like reasonable price to pay to allow "Getting Started" snippets to be painless and friendly.